vllm/executor/executor_base.py

```diff
@@ -18,6 +18,8 @@ class ExecutorBase(ABC):
     that can execute the model on multiple devices.
     """
 
+    uses_ray: bool  # whether the executor uses Ray for orchestration.
```

Comment:
Consider making it a property abstract class? That way, every subclass should implement this?

Reply:
this should already function like that (it will raise exceptions if not set in implementation)